analyzers/src/SonarAnalyzer.CSharp/Rules/ExceptionShouldNotBeThrownFromUnexpectedMethods.cs

```diff
@@ -36,14 +37,13 @@ namespace SonarAnalyzer.Rules.CSharp
     public sealed class ExceptionShouldNotBeThrownFromUnexpectedMethods : SonarDiagnosticAnalyzer
     {
         internal const string DiagnosticId = "S3877";
-        private const string MessageFormat = "Remove this 'throw' statement.";
+        private const string MessageFormat = "Remove this 'throw' {0}.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
```

Comment:
we should reorder the properties by their visibility.  `Rule` is private, then `SupportedDiagnostics` public and then again `DefaultAllowedExceptions` is private.